node/tools/src/bin/deployer.rs

```diff
@@ -0,0 +1,150 @@
+//! Deployer for the kubernetes cluster.
+use std::net::SocketAddr;
+use std::str::FromStr;
+use std::{fs, path::PathBuf};
+
+use anyhow::Context;
+use clap::{Parser, Subcommand};
+use rand::Rng;
+use zksync_consensus_crypto::{Text, TextFmt};
+use zksync_consensus_roles::node::SecretKey;
+use zksync_consensus_tools::k8s;
+use zksync_consensus_tools::AppConfig;
+use zksync_consensus_tools::NodeAddr;
+
+const NAMESPACE: &str = "consensus";
+const NODES_PORT: u16 = 3054;
+
+/// Command line arguments.
+#[derive(Debug, Parser)]
+#[command(name = "deployer")]
+struct DeployerCLI {
+    /// Subcommand to run.
+    #[command(subcommand)]
+    command: DeployerCommands,
+}
+
+/// Subcommand arguments.
+#[derive(Debug, Parser)]
+struct SubCommandArgs {
+    /// Number of nodes to deploy.
+    #[arg(long)]
+    nodes: usize,
+}
+
+/// Subcommands.
+#[derive(Subcommand, Debug)]
+enum DeployerCommands {
+    /// Generate configs for the nodes.
+    GenerateConfig(SubCommandArgs),
+    /// Deploy the nodes.
+    Deploy(SubCommandArgs),
+}
+
+/// Generates config for the nodes to run in the kubernetes cluster
+/// Creates a directory for each node in the parent k8s_configs directory.
+fn generate_config(nodes: usize) -> anyhow::Result<()> {
```

Comment:
Yeah we discussed this. But opted to reutilize the already existing load-test config script. And also I thought the command line would be too long for the whole contents of the config file + node key + validator key.
Still its a good idea to implement this to avoid the three steps (`create configs` -> `build images` -> `deploy`) workflow

Reply:
It doesn't make sense to build images per node/config (or even just one, but built after the config is generated). There should be only one image for the node, re-built on non-test code changes. Config should be applied only when spawning the container. If it's not possible to do so using a config file, then it's another reason to not use it. (in addition to other complexities it introduces)

To support this, the node should be capable of running without a config file, applying default/zero values. If config file exists, it should be applied. Then, if any CLI flags were provided, they should override the corresponding config. For required fields, either the config file or the respective flag must be provided.

Reply:
Created a sub-issue for this